i3_resurrect/util.py

```diff
@@ -1,8 +1,27 @@
 import sys
-
+import string
 
 def eprint(*args, **kwargs):
     """
     Function for printing to stderr.
     """
     print(*args, file=sys.stderr, **kwargs)
+
+
+def filename_filter(filename):
+    """
+    Take a string and return a valid filename constructed from the string.
+"""
+    blacklist = '/\:*"<>|'
+    # Alternatively use whitelist approch
+    # whitelist = "-_.() %s%s" % (string.ascii_letters, string.digits)
+    if filename is None:
+        return filename
+
+    for char in filename:
```

Comment:
Mightn't [re.sub()](https://docs.python.org/3.5/library/re.html#re.sub) be a more concise way to remove the characters? That way you could do it in one line instead of looping through characters.

Reply:
I found string.replace() command simpler to use with special characters.
kept  it but used more concise expression.
If you have better solution you can replace this part of code instead.

Reply:
Fair enough, regular expressions can be horribly unreadable a lot of the time anyway :smile: